### DISCUSSION TITLE
What is PQ good for?

### DISCUSSION THREAD

**Discussion by novellizator** [vitaly-t/pg-promise#844] [General]
So I usually run stuff like this:

> await db.map(query, [param1])

Where the **query** can either be a string or a [PQ](http://vitaly-t.github.io/pg-promise/ParameterizedQuery.html) . What are the advantages of using PQ? 
The disadvantages are clear, for example if the query in PQ contains `$1:list` I get a syntax error. Can I drop PQ() from the whole codebase without any disadvantage? What should I look out for?

**Comment by vitaly-t** [comment]
Prepared Queries are formatted on the server side. And they can be additionally optimized for performance, i.e. SQL execution plan is cached.
